decode.go

```diff
@@ -30,8 +32,8 @@ func decodedLen(src []byte) (blockLen, headerLen int, err error) {
 	if n <= 0 {
 		return 0, 0, ErrCorrupt
 	}
-	if uint64(int(v)) != v {
-		return 0, 0, errors.New("snappy: decoded block is too large")
+	if uint64(uint32(v)) != v {
```

Comment:
Yes, it should be > not >=.

I don't think it needs explicit documentation. ioutil.ReadFile will fail if it tries to read a 5G file on a 32-bit platform.

Still, it does need to be explicitly checked for, and return different errors for "too large to represent on this computer" and "invalid on any computer".